jobrunner/tracing.py

```diff
@@ -0,0 +1,226 @@
+import logging
+import os
+
+from opentelemetry import trace
+from opentelemetry.exporter.otlp.proto.http.trace_exporter import OTLPSpanExporter
+from opentelemetry.sdk.trace import TracerProvider
+from opentelemetry.sdk.trace.export import ConsoleSpanExporter, SimpleSpanProcessor
+from opentelemetry.trace.propagation.tracecontext import TraceContextTextMapPropagator
+
+from jobrunner import config
+from jobrunner.lib import database
+from jobrunner.models import Job, SavedJobRequest, StatusCode
+
+
+logger = logging.getLogger(__name__)
+provider = TracerProvider()
+trace.set_tracer_provider(provider)
+
+
+def add_exporter(exporter, processor=SimpleSpanProcessor):
+    # we default to SimpleSpanProcessor so that it's synchronous for tests
+    provider.add_span_processor(processor(exporter))
+
+
+def setup_default_tracing():
+    """Inspect environmenet variables and set up exporters accordingly.
+
+    We use the SimpleSpanProcessor by default, which is synchronous. The
+    BatchSpanProcess is asynchronous and more efficent at scale, but are
+    emitting so few spans that it is easier just to keep it simpler, and it
+    makes testing easier.
+    """
+    if "OTEL_EXPORTER_OTLP_HEADERS" in os.environ:
+        if "OTEL_SERVICE_NAME" not in os.environ:
+            raise Exception(
+                "OTEL_EXPORTER_OTLP_HEADERS is configured, but missing OTEL_SERVICE_NAME"
+            )
+        if "OTEL_EXPORTER_OTLP_ENDPOINT" not in os.environ:
+            os.environ["OTEL_EXPORTER_OTLP_ENDPOINT"] = "https://api.honeycomb.io"
```

Comment:
open question - should this default to functioning (honeycomb url) or safe (invalid url)? for TPP we're confident that if it defaults to honeycomb the firewall will block it, but is that an assumption we can lean on in future?

Reply:
It's a fair question.

I think given that we're happy for any such data to live in honeycomb eventually, I think it's ok?

It makes configuring local runs to emit to honeycomb a bit easier, is the main benefit.

Reply:
yep ok